crates/subspace-farmer/src/bin/subspace-farmer/commands/farm.rs

```diff
@@ -35,15 +36,19 @@ pub(crate) async fn farm(
     info!("Connecting to node at {}", node_rpc_url);
     let client = NodeRpcClient::new(&node_rpc_url).await?;
 
+    let metadata = client
+        .farmer_metadata()
+        .await
+        .map_err(|error| anyhow!(error))?;
+    let max_plot_size = max_plot_size
+        .map(|max_plot_size| max_plot_size / PIECE_SIZE as u64)
+        .unwrap_or(metadata.max_plot_size);
```

Comment:
`max_plot_size` from CLI should be used as an additional restriction, not as an override (because if you make it bigger you'll make some of the solutions fail due to pieces outside of allowed range).

Reply:
Should I throw an error here if the user specifies some bigger amount than in the consensus?

Reply:
I think logging warning is sufficient